### PR TITLE
feat: upgrade to angular 10

### PR DESCRIPTION
**Pull request by drewf7**
Hi,

I ended up going down a bit of a rabbit hole with this one, but included in this PR are (I think) all the changes/updates needed to be compatible with Angular 10, Typescript 4.0, and NodeJS 12.

We also fixed `616 vulnerabilities (267 low, 129 moderate, 220 high) `.

Since this will likely break compatibility with older versions of Angular, I think a new major version number is probably needed. I also added a section to the readme to call out this break.

**Testing Done**
I ran your `npm test` script. That weeded out a couple of extra changes that were needed for the webpack stuff, and new dependency requirements for ng-packagr.

I ran `npm start` and verified that the demo page was loading.

Finally I ran `npm run build:dist` and replaced the copy of anguar-gauge running in my Angular 10 app with the fresh build, then verified it loaded and worked as expected.

I know this is a good chunk of changes, so I'd be happy to do other testing if you want.

And apologies for the line ending issues in the commits. There are also some weird spacing things in a few of the files due to the use of `ng update` to apply some automatic migrations.

Thanks

### CODE REVIEW

**Comment by josephdt12** [comment]
Seems like a great PR! Please approve

File: package.json [hunk 1]
```diff
@@ -56,69 +55,83 @@
...
-    "zone.js": "^0.8.20"
+    "webpack-cli": "^3.3.12",
+    "webpack-dev-server": "^3.11.0",
+    "webpack-rxjs-externals": "^2.0.0",
+    "zone.js": "^0.11.1"
   },
   "peerDependencies": {
-    "@angular/core": ">=5.0.0"
+    "@angular/core": ">=10.0.0",
+    "svg-gauge": "^1.0.6"
```
Comment (mattlewis92) [inline package.json]: Can we keep `svg-gauge` out of peer dependencies? Just one less thing for people to worry about in their package.json as we already install it for them as a dependency.

**Review by mattlewis92 — CHANGES_REQUESTED** [review]
Thanks so much for taking the time to do this! Just one small comment about removing the dependency from peer dependencies and we should be good to merge 😄

File: README.md [hunk 2]
```diff
@@ -22,6 +23,13 @@ https://mattlewis92.github.io/angular-gauge/
 
 A tiny angular 5.0+ wrapper for https://github.com/naikus/svg-gauge
```
Comment (mattlewis92) [inline README.md]: Oh also can you change this to `angular 10+`

**Comment by mattlewis92** [comment]
Was just looking at the console output from the tests, if you want to get rid of those warnings from webpack you can add these 3 changes to `karma.conf.ts`:
https://github.com/mattlewis92/angular-resizable-element/blob/master/karma.conf.ts#L23
https://github.com/mattlewis92/angular-resizable-element/blob/master/karma.conf.ts#L66-L69
https://github.com/mattlewis92/angular-resizable-element/blob/master/karma.conf.ts#L53-L58

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "angular-gauge",
-  "version": "3.1.3",
+  "version": "4.0.0",
```
Comment (mattlewis92) [inline package.json]: Can you also change this back to `"version": "3.1.3",` as it gets bumped automatically on release.

File: package.json (hunk 1 above)
Comment (drewf7) [reply package.json]: Good call. Updated.

File: README.md (hunk 2 above)
Comment (drewf7) [reply README.md]: Done!

File: package.json (hunk 3 above)
Comment (drewf7) [reply package.json]: Done!

**Comment by drewf7** [comment]
Hi @mattlewis92 Thanks for taking the time to go through this.

I've made the updates you suggested. Sorry for all the line ending issues. I'm really having a time with those.

I tried commiting `git add --renormalize .` but that actually made things worse.

Definitely open to suggestions there.

Happy to make any other requested changes.

Thanks!

**Review by mattlewis92 — APPROVED** [review]
LGTM, thank you for taking this on! 🙌

**Comment by intotecho** [comment]
Not sure where the New Issue button in....
I noticed that [color] and some other input directives are not in the README/user guide. 
It would also help if the README made it clearer that classes have to be declared in **global** styles. I put the <mwl-gauge> styles into my component's CSS and it didn't work.  Things like this must go into the global style sheet, since Angular runs style encapsulation per component, which means that a style defined in a component's css will only be applied to its own template (and never to others). For more information, see [style-scope.](https://angular.io/guide/component-styles#style-scope)

An example angular component/app in StackBliz would be great. To change the color from the component, I had to do this as gaugeColor() wasn't binding to 'this'.  

TS
```
@Component({
  selector: 'my-comp',
  templateUrl: './my-comp.html',
  styleUrls: ['./my-comp.scss']
})
export class MyComponent implements OnInit {

public gaugeColor(val: number) {  
    if(this != undefined) {
      return (this.myState) 'red': 'blue';
    }
    return 'green';
  }

ngOnInit(): void {
    this.gaugeColorFunction = this.gaugeColor.bind( this );
}
```
Template
```
  <mwl-gauge
        ....
        [color]="gaugeColorFunction"
        >
    </mwl-gauge>

```

**Pull request merged**